AuthenticationAndAuthorization/AuthenticationAndAuthorization/Endpoints/GenreEndpoints.cs

```diff
@@ -0,0 +1,23 @@
+﻿using AuthenticationAndAuthorization.Policy;
+using BusinessLogic.Interfaces;
+using BusinessLogic.Model.GenreModel;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AuthenticationAndAuthorization.Endpoints
+{
+    public static class GenreEndpoints
+    {
+        public static void MapGenreEndpoints(this IEndpointRouteBuilder endpoints)
+        {
+            endpoints.MapPost("create/genre", ([FromServices] IGenreService genreService, [FromBody] CreateGenreModel genre)
+                => genreService.CreateGenre(genre)).RequireAuthorization(Policies.AdminPolicy);
+
+            endpoints.MapPost("delete/genre", ([FromServices] IGenreService genreService, [FromBody] DeleteGenreModel genre)
+                => genreService.DeleteGenre(genre)).RequireAuthorization(Policies.AdminPolicy);
+
+            endpoints.MapGet("get/genre", ([FromServices] IGenreService genreService)
+                => genreService.GetGenre()).RequireAuthorization(Policies.UserPolicy);
```

Comment:
Follow REST API conventions

```suggestion
            endpoints.MapPost("genres", ([FromServices] IGenreService genreService, [FromBody] CreateGenreModel genre)
                => genreService.CreateGenre(genre)).RequireAuthorization(Policies.AdminPolicy);

            endpoints.MapDelete("genres/{id:guid}", ([FromServices] IGenreService genreService, [FromBody] DeleteGenreModel genre)
                => genreService.DeleteGenre(genre)).RequireAuthorization(Policies.AdminPolicy);

            endpoints.MapGet("genres", ([FromServices] IGenreService genreService)
                => genreService.GetGenre()).RequireAuthorization(Policies.UserPolicy);
```